src/dash-table/derived/paginator.ts

```diff
@@ -61,21 +109,60 @@ function getFrontEndPagination(
 
             page_current--;
             setProps({ page_current, ...clearSelection });
+        },
+        loadFirst: () => {
+            page_current = 0;
+            setProps({ page_current, ...clearSelection });
+        },
+        loadLast: () => {
+            page_current = lastPage(data, page_size);
+            setProps({ page_current, ...clearSelection });
+        },
+        lastPage: lastPage(data, page_size),
+        goToPage: (page_number: string) => {
+
+            let page = parseInt(page_number, 10);
+
+            if (isNaN(page)) {
+                return;
+            }
+
+            page--;
+
+            if (page < 0) {
+                page_current = 0;
+                setProps({ page_current, ...clearSelection });
+                return;
+            }
+
+            if (page > lastPage(data, page_size)) {
+                page_current = lastPage(data, page_size);
+                setProps({ page_current, ...clearSelection });
+                return;
+            }
+
+            page_current = page;
+            setProps({ page_current, ...clearSelection });
         }
     };
 }
 
 function getNoPagination() {
     return {
         loadNext: () => { },
-        loadPrevious: () => { }
+        loadPrevious: () => { },
+        loadFirst: () => { },
+        loadLast: () => { },
+        lastPage: 0,
+        goToPage: () => { }
```

Comment:
The paginator could be responsible for determining the values used to enable/disable the UI: hasPrevious, hasNext, hasLast, keeping all the pagination logic together instead of separating it between the helper and the UI.